Car.cpp

```diff
@@ -1,7 +1,54 @@
-#include "Car.hpp"
-#include <iostream>
-
-void Car::turnLeft()      { std::cout << __FUNCTION__ << std::endl; }
-void Car::turnRight()     { std::cout << __FUNCTION__ << std::endl; }
-void Car::brake()         { std::cout << __FUNCTION__ << std::endl; }
-void Car::accelerate(int) { std::cout << __FUNCTION__ << std::endl; }
+#include "Car.hpp"
+#include <iostream>
+#include "Exceptions.hpp"
+#include <string>
+
+void Car::turnLeft()         { std::cout << __FUNCTION__ << std::endl; }
+void Car::turnRight()  	     { std::cout << __FUNCTION__ << std::endl; }
+void Car::brake()           
+{
+    velocity = 0;
+}
+
+void Car::accelerate(int acceleration)
+{ 
+    if(acceleration == 0)
+    {
+        return;
+    }
+
+    if(acceleration < 0)
+    {        
+        throw InvalidAccelerationValue(
+            std::to_string(acceleration));
+    }
+    else    
+    {
+        velocity += acceleration;
+    }
+}
+
+void Car::changeGear(Gear gear)
+{
+    if(currentGear == gear)
+    {
+        return;
+    }
+    
+    if(currentGear > Gear::_N 
+        && gear == Gear::_R)
+    {
+        throw InvalidGear();
+    }
+    currentGear = gear;
+}
```

Comment:
Sam koncept oddzielnego typu na biegi jest mega fajny. Tylko czy nie powinny one być w PetrolCar? Z założenia ElectricCar miał w ogóle nie mieć biegów, bo chyba silniki elektryczne biegów nie potrzebują. Być może w niektórych można emulować takie zachowanie jak biegi.  Tutaj brakuje mi też wywołania funkcji `changeGear(int)` z silnika spalinowego. Dodaliście nowe pole Gear w klasie Car i go używacie, podczas gdy bieg będzie raczej własnością silnika (może właściwie to skrzyni biegów, ale uznajmy, że ona należy do silnika).

Reply:
Zgadzam się z tym w pełni ale uznałem, że pewne rzeczy możemy uprościć ze względu na to, że inny projekt czeka w kolejce :D

Reply:
ok :)